Update README with newly created OperatorSource

anik120 opened the pull request:
Use upstream-community-operators OperatorSource for installation on k8s.

aravindhp commented on README.md:
```diff
@@ -28,7 +28,7 @@ $ export REGISTRY=<SOME_REGISTRY> \
 ### Description
 The operator manages two CRDs: [OperatorSource](./deploy/crds/operatorsource.crd.yaml) and [CatalogSourceConfig](./deploy/crds/catalogsourceconfig.crd.yaml).
 
-`OperatorSource` is used to define the external datastore we are using to store operator bundles. At the moment we only support Quay's app-registry as our external datastore. Please see [here](deploy/examples/operatorsource.cr.yaml) for an example `OperatorSource`. The `endpoint` in the `spec` is typically set to `https:/quay.io/cnr` if you are using Quay's app-registry. The `registryNamespace` i ...
```
You are removing the documentation for private Quay repos. Please add that back.

aravindhp commented on README.md:
```diff
@@ -48,18 +48,18 @@ $ kubectl apply -f deploy/upstream
 
 #### Installing an operator using Marketplace
-The following section assumes that Marketplace was installed in the `openshift-marketplace` namespace. For Marketplace to function you need to have at least one `OperatorSource` CR present on the cluster. To get started you can use the [community](deploy/examples/operatorsource.cr.yaml) OperatorSource. If you are on an OKD cluster, you can skip this step as it is installed by default.
+The following section assumes that Marketplace was installed in the `openshift-marketplace` namespace. For Marketplace to function you need to have at least one `OperatorSource` CR present on the cluster. To get started you can use the [upstream-community-operators](deploy/examples/operatorsource.cr.yaml) OperatorSource. If you are on an OKD cluster, you can skip this step as it is installed by d ...
```
Given you are targeting this for upstream, please update the namespace to `marketplace`. It is better to say `use the OperatorSource` for upstream community operators. Please use `` when referring to `OperatorSource`.

Also update what is installed by default on an OKD cluster.

aravindhp commented on README.md:
```diff
@@ -48,18 +48,18 @@ $ kubectl apply -f deploy/upstream
-community-operators   federationv2,svcat,metering,etcd,prometheus,automationbroker,templateservicebroker,cluster-logging,jaeger,descheduler
+$ kubectl get opsrc upstream-community-operators -o=custom-columns=NAME:.metadata.name,PACKAGES:.status.packages -n openshift-marketplace
+NAME                           PACKAGES
+upstream-community-operators   jaeger
```
Can we show multiple packages here?

anik120 replied:
@aravindhp we could once we have multiple operators in the upstream registry. For now we only have jaeger in the registry.

aravindhp replied:
These are just showing examples. It doesn't need to reflect what is exactly there in the repository.

aravindhp commented on README.md:
````diff
@@ -48,18 +48,18 @@ $ kubectl apply -f deploy/upstream
 ```
 
-Now if you want to install the `descheduler` and `jaeger` operators, create a `CatalogSourceConfig` CR as shown below:
+Now if you want to install the `jaeger` operator, create a `CatalogSourceConfig` CR as shown below:
````
I would like to use this example to call out that you use the same `CatalogSourceConfig` to install multiple operators.

anik120 replied:
We don't have descheduler or any other operator besides jaeger in the `upstream-community-operator` OperatorSource. We could update the README to reflect the point above as soon as we have another operator in the `upstream-community-operator` OperatorSource.

aravindhp replied:
Again, we are showing examples here. It doesn't need to reflect exactly what is present in the repository which is going to be changing. We are not going to keep updating the README each time the Quay repo changes.

SamiSousa commented on deploy/examples/operatorsource.cr.yaml:
```diff
@@ -1,11 +1,11 @@
-  name: community-operators
-  namespace: openshift-marketplace
+  name: upstream-community-operators
+  namespace: marketplace
```
Should this be `openshift-marketplace`?

anik120 replied:
The section starts with 
> The following section assumes that Marketplace was installed in the openshift-marketplace namespace

I suppose it makes more sense to replace all instances of `openshift-marketplace` with just `marketplace`

SamiSousa commented on deploy/examples/operatorsource.cr.yaml:
```diff
@@ -1,11 +1,11 @@
-  registryNamespace: community-operators
-  displayName: "Community Operators"
+  registryNamespace: upstream-community-operators
+  displayName: "Upstream Community Operators"
```
The readme inline example CSC doesn't match this display name

aravindhp commented on deploy/examples/operatorsource.cr.yaml:
```diff
@@ -1,11 +1,11 @@
 metadata:
-  name: community-operators
-  namespace: openshift-marketplace
+  name: upstream-community-operators
```
Please create a new file for this `OperatorSource`.

anik120 commented:
@SamiSousa @aravindhp addressed comments in latest commit. Please review. 

Note: I created a new file called upstream.operatorsource.cr.yaml and renamed the existing operatorsource.cr.yaml to community.operatorsource.cr.yaml

aravindhp commented on README.md:
```diff
@@ -48,43 +48,43 @@ $ kubectl apply -f deploy/upstream
-  packages: descheduler,jaeger
-  csDisplayName: "Community Operators"
+  targetNamespace: local-operators
+  packages: descheduler, jaeger
```
No space between the packages

anik120 commented:
@aravindhp Addressed. Also https://github.com/operator-framework/community-operators/pull/86

SamiSousa commented on deploy/examples/upstream.operatorsource.cr.yaml:
```diff
@@ -0,0 +1,11 @@
+apiVersion: marketplace.redhat.com/v1alpha1
```
Should we keep both example `OperatorSource` CRs? Having both might suggest that both should be applied, which I don't think we want.

anik120 replied:
@SamiSousa https://github.com/operator-framework/operator-marketplace/pull/112#discussion_r257819134

I think we decided to keep both examples and direct users to the appropriate example in the appropriate section/discussion.

SamiSousa commented on README.md:
```diff
@@ -48,50 +48,51 @@ $ kubectl apply -f deploy/upstream
-  source: installed-community-operators
+  name: jaeger
+  source: installed-upstream-community-operators
+  sourceNamespace: local-operators
```
:+1:

aravindhp commented:
/retest

aravindhp commented:
/retest

anik120 commented:
/test e2e-aws

anik120 commented:
/retest

anik120 commented:
/retest

anik120 commented:
/retest
